bootstore/src/schemes/v0/peer.rs

```diff
@@ -0,0 +1,797 @@
+    ///
+    /// These are generated from DDM prefixes learned by the bootstrap agent.
+    PeerAddresses(BTreeSet<SocketAddrV6>),
+
+    /// Get the status of this peer
+    GetStatus { responder: oneshot::Sender<Status> },
+
+    /// Shutdown the peer
+    Shutdown,
+}
+
+// A handle for interacting with a `Peer` task
+pub struct PeerHandle {
+    tx: mpsc::Sender<PeerApiRequest>,
+}
+
+impl PeerHandle {
+    /// Initialize a rack at the behest of RSS running on the same scrimlet as
+    /// this Peer
+    pub async fn init_rack(
+        &self,
+        rack_uuid: RackUuid,
+        initial_membership: BTreeSet<Baseboard>,
+    ) -> Result<(), PeerRequestError> {
+        let (tx, rx) = oneshot::channel();
+        self.tx
+            .send(PeerApiRequest::InitRack {
+                rack_uuid: RackUuid(Uuid::new_v4()),
+                initial_membership,
+                responder: tx,
+            })
+            .await?;
+        let res = rx.await?;
+        res
+    }
+
+    /// Initialize this peer as a learner
+    pub async fn init_learner(&self) -> Result<(), PeerRequestError> {
+        let (tx, rx) = oneshot::channel();
+        self.tx.send(PeerApiRequest::InitLearner { responder: tx }).await?;
+        let res = rx.await?;
+        res
+    }
+
+    /// Load the rack secret.
+    ///
+    /// This can only be successfully called when a peer has been initialized,
+    /// either as initial member or learner who has learned its share.
+    pub async fn load_rack_secret(
+        &self,
+    ) -> Result<RackSecret, PeerRequestError> {
+        let (tx, rx) = oneshot::channel();
+        self.tx.send(PeerApiRequest::LoadRackSecret { responder: tx }).await?;
+        let res = rx.await?;
+        res
+    }
+
+    /// Inform the peer of currently known IP addresses on the bootstrap network
+    ///
+    /// These are generated from DDM prefixes learned by the bootstrap agent.
+    pub async fn load_peer_addresses(
+        &self,
+        addrs: BTreeSet<SocketAddrV6>,
+    ) -> Result<(), PeerRequestError> {
+        self.tx.send(PeerApiRequest::PeerAddresses(addrs)).await?;
+        Ok(())
+    }
+
+    /// Get the status of this peer
+    pub async fn get_status(&self) -> Result<Status, PeerRequestError> {
+        let (tx, rx) = oneshot::channel();
+        self.tx.send(PeerApiRequest::GetStatus { responder: tx }).await?;
+        let res = rx.await?;
+        Ok(res)
+    }
+
+    /// Shutdown the peer
+    pub async fn shutdown(&self) -> Result<(), PeerRequestError> {
+        self.tx.send(PeerApiRequest::Shutdown).await?;
+        Ok(())
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct Status {
+    fsm_state: &'static str,
+    peers: BTreeSet<SocketAddrV6>,
+    connections: BTreeMap<Baseboard, SocketAddrV6>,
+    accepted_connections: BTreeSet<SocketAddrV6>,
+    negotiating_connections: BTreeSet<SocketAddrV6>,
+}
+
+/// A peer in the bootstore protocol
```

Comment:
A comment about how this struct is used would be helpful.

Do we have one `Peer` per known other sled doing bootstore?  Or do we only have one `Peer`, representing the sled that this code is running on?